.buildkite/scripts/steps/code_generation/security_solution_codegen.sh

```diff
@@ -4,7 +4,14 @@ set -euo pipefail
 
 source .buildkite/scripts/common/util.sh
 
+echo --- Security Solution OpenAPI linting
+
+cd x-pack/plugins/security_solution
+
+yarn openapi:lint
+
 echo --- Security Solution OpenAPI Code Generation
 
-(cd x-pack/plugins/security_solution && yarn openapi:generate)
-check_for_changed_files "yarn openapi:generate" true
\ No newline at end of file
+yarn openapi:generate
```

Comment:
If we always lint before generating (which makes sense), why don't we just do it in the generator itself? This would give a couple of benefits:

- Other teams will get OAS linting out of the box.
- We won't have duplication, such as `"openapi:lint": "npx @redocly/cli lint --config ../../../packages/kbn-openapi-generator/redocly_linter/config.yaml $(find ./ -name '*.schema.yaml')",` in `x-pack/plugins/osquery/package.json`
- The command for running linting will be defined in the same place where we have the `redocly` config.

We could do linting out of the box by default + skip it if specified via params + expose a dedicated function for linting for those who would like to have it as a separate step.

Reply:
Integrated linter command into generator [here](https://github.com/elastic/kibana/pull/171851/commits/0b7fe492ecdf2f979fcdb01d1d7e91d94aff9277).